lsp/src/main/java/org/aya/lsp/options/RenderOptions.java

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.lsp.options;
+
+import kala.collection.Seq;
+import kala.collection.immutable.ImmutableMap;
+import kala.collection.mutable.MutableMap;
+import kala.control.Option;
+import kala.control.Result;
+import org.aya.lsp.models.ServerOptions;
+import org.aya.pretty.backend.html.DocHtmlPrinter;
+import org.aya.pretty.backend.latex.DocTeXPrinter;
+import org.aya.pretty.doc.Doc;
+import org.aya.pretty.printer.ColorScheme;
+import org.aya.pretty.printer.StyleFamily;
+import org.aya.pretty.style.AyaColorScheme;
+import org.aya.pretty.style.AyaStyleFamily;
+import org.javacs.lsp.MarkupKind;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @param colorScheme
+ * @param styleFamily
+ */
+public record RenderOptions(
+  @NotNull Option<ColorScheme> colorScheme,
+  @NotNull Option<StyleFamily> styleFamily
+) {
+  public final static RenderOptions DEFAULT = new RenderOptions(
+    Option.some(AyaColorScheme.EMPTY),
+    Option.some(AyaStyleFamily.DEFAULT)
+  );
+
+  public enum RenderTarget {
+    Debug,
+    HTML,
+    TeX
+  }
+
+  public static final @NotNull ImmutableMap<String, ColorScheme> BUILTIN_COLOR_SCHEMES = ImmutableMap.of(
+    "emacs", AyaColorScheme.EMACS,
+    "intellij", AyaColorScheme.INTELLIJ,
+    "none", AyaColorScheme.EMPTY
+  );
+
+  /// region Helper
+
+  /**
+   * Construct a {@link RenderOptions} from {@link ServerOptions}
+   */
+  @Contract(pure = true)
+  public static @NotNull Result<@NotNull RenderOptions, @NotNull String> fromServerOptions(@NotNull ServerOptions options) {
+    Option<ColorScheme> colorScheme = Option.none();
+
+    // colorScheme is none if both colorName and colorOverride are null
+    if (options.colorName != null || options.colorOverride != null) {
```

Comment:
I see `parseColorScheme` produces `EMPTY` when `colorName` is not set. Then it should be the same here --- `colorName` is not set. 

In particular, use `EMPTY` here so:
- the field `RenderOptions#colorScheme` does not need to live in an `Option`
- some `getOrDefault` checks in `renderXXX` can be deleted.

If we do the same to `RenderOptions#styleFamily`, the `RenderOptions.DEFAULT` can be deleted.